### PR TITLE
4.2 Add Turtle Mode

### PR DESCRIPTION
**Pull request by Hwurzburg**

### CODE REVIEW

**Comment by Hwurzburg** [comment]
@andyp1per this needs a review ASAP for merging...thanks

File: copter/source/docs/turtle-mode.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+.. _turtle-mode:
+
+===========
+Turtle Mode
+===========
+
+Turtle Mode is a special mode that can be evoked either as a :ref:`flight mode <flight-modes>` via the flight mode switch or as an :ref:`Auxilary function <common-auxiliary-functions>` assigned to an RC switch (channel). Turtle mode allows a user to attempt to flip the Copter upright, if inverted, by reversing the direction of adjacent pairs of motors and producing thrust to un-invert the vehicle after a crash.
```
Comment (andyp1per) [inline copter/source/docs/turtle-mode.rst:7]: "invoked"

File: copter/source/docs/turtle-mode.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+DShot capable ESCs are required for this mode to function, allowing the DShot reverse command to be sent to the ESCs.
```
Comment (andyp1per) [inline copter/source/docs/turtle-mode.rst:9]: You need to mention that SERVO_DSHOT_ESC = 1 also needs to be set

File: copter/source/docs/turtle-mode.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+Once in this mode, moving the roll and/or pitch stick of the transmitter away from center will reverse and spin up the designated pair of motors, increasing thrust to maximum at full stick throw.
```
Comment (andyp1per) [inline copter/source/docs/turtle-mode.rst:11]: might want to advise doing this for any extended period may damage the motors

**Review by andyp1per — CHANGES_REQUESTED** [review]

**Pull request merged**